backend/src/middlewares/adminAudit.ts

```
@@ -0,0 +1,41 @@
+import { Request, Response, NextFunction } from 'express';
+import { AdminAuditService } from '@/services/AdminAuditService';
+
+/**
+ * Middleware: records a minimal admin audit event for write operations under /api/admin
+ * Captures method, route, userId, role, and outcome status. No PII.
+ */
+export function adminAudit(req: Request, res: Response, next: NextFunction): void {
+  // Only record for mutating methods
+  if (!['POST', 'PUT', 'PATCH', 'DELETE'].includes(req.method)) {
+    return next();
```

Biome (2.1.2)

[error] The function should not return a value because its return type is void.

The function is here:

'void' signals the absence of value. The returned value is likely to be ignored by the caller.

(lint/correctness/noVoidTypeReturn)